src/App.jsx

```diff
@@ -1,5 +1,130 @@
+import { useEffect, useState } from 'react';
+import { Global, css } from '@emotion/react';
+import styled from '@emotion/styled';
+import InputForm from './components/InputForm.jsx';
+import TodoList from './components/TodoList.jsx';
+import Header from './components/Header.jsx';
+import bgImage from './grid.jpg';
+
+const GlobalStyles = () => (
+  <Global
+    styles={css`
+      @font-face {
+        font-family: 'RixXladywatermelonR';
+        src: url('https://fastly.jsdelivr.net/gh/projectnoonnu/2408-4@1.0/RixXladywatermelonR.woff2')
+          format('woff2');
+        font-weight: normal;
+        font-style: normal;
+      }
+
+      html {
+        margin: 0;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        height: 100vh;
+      }
+
+      body {
+        font-family: 'RixXladywatermelonR', sans-serif;
+        max-width: 500px;
+        padding: 10px;
+        text-align: center;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        scrollbar-width: thin;
+        margin: 0;
+      }
+
+      button {
+        font-family: 'RixXladywatermelonR';
+        display: flex;
+        justify-content: center;
+        align-content: center;
+        flex-wrap: wrap-reverse;
+      }
+    `}
+  />
+);
+
+const ContainerStyle = styled.div`
+  position: relative;
+  display: flex;
+  width: 450px;
+  height: 680px;
+  flex-direction: column;
+  background-color: #fef79fbd;
+  overflow: hidden;
+  ::after {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background-image: url(${bgImage});
+    background-size: cover;
+    background-blend-mode: darken;
+    opacity: 0.5;
+    z-index: -1;
+  }
+`;
+
 function App() {
-  return <div>React Todo</div>;
+  const userName = '옹헤';
+  //localStorage에서 todoList 불러오기
+  const [todoList, setTodoList] = useState(() => {
+    const prevTodoList = localStorage.getItem('todoList');
+    return prevTodoList ? JSON.parse(prevTodoList) : [];
+  });
+
+  //todoList가 변경될 때마다 localStorage에 저장
+  useEffect(() => {
+    localStorage.setItem('todoList', JSON.stringify(todoList));
+  }, [todoList]);
+
+  //사용자에게 받은 입력으로 todo 추가하기
+  function handleAddTodo(todoInput) {
+    const newTodo = {
+      createTime: Date.now(),
+      content: todoInput,
+      isCompleted: false,
+    };
+    setTodoList([...todoList, newTodo]);
```

Comment:
스프레드 연산자를 활용해서 추가하는 방법이 훨 간단하고 보기에도 편하네요👍

Reply:
`spread` 연산자 사용 아주 좋습니다! 다만 `setState`를 사용할 때 변경되는 state를 직접 참조하여 사용하는 것보다는 콜백 함수를 작성하는 방식이 조금 더 안정적인 방식입니다 🙂

```suggestion
    setTodoList((prev) => [...prev, newTodo]);
```

---

_그 이유가 궁금하다면..._

프로젝트를 진행하다 보면 `state 변경 -> 렌더링 -> state 변경 -> 렌더링 -> ...` 과정을 반복하게 됩니다.

`setState(state + 1)`은 현재의 state 값을 기반으로 상태를 업데이트 합니다. 하지만 이 방식은 state 값이 변경되는 과정에서 다른 setState 호출이 있을 경우, 이전 렌더링 시점의 상태를 기반으로 하기 때문에 최신 상태값을 반영하지 못할 수 있습니다.

반면 `setState(prev => prev + 1)`은 콜백 함수를 사용하여 이전 상태(prev)를 기준으로 상태를 업데이트 합니다. 이 방법은 상태 업데이트가 비동기적으로 처리되는 동안에도 항상 최신 상태를 반영하므로, 상태가 여러 번 업데이트 될 때에도 정확한 결과를 보장한다고 합니다!